### PR TITLE
Cr 3489 container cluster

### PR DESCRIPTION
**Pull request by doronafota**

### CODE REVIEW

File: cloudrail/knowledge/context/gcp/resources/cluster/gcp_container_cluster.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    @property
+    def is_tagable(self) -> bool:
+        return False
+
+    @property
+    def is_labeled(self) -> bool:
+        return False
+
+    def get_id(self) -> str:
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/context/gcp/resources/cluster/gcp_container_cluster.py:75]: Do not use get_id, use get_name instead.

File: cloudrail/knowledge/context/gcp/resources_builders/terraform/container_cluster_builder.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class ContainerClusterBuilder(BaseGcpTerraformBuilder):
+
+    def do_build(self, attributes: dict) -> GcpContainerCluster:
+        name = attributes["name"]
+        location = self._get_known_value(attributes, "location")
+        cluster_ipv4_cidr = self._get_known_value(attributes, "cluster_ipv4_cidr")
+        enable_shielded_nodes = attributes["enable_shielded_nodes"]
+        master_authorized_networks_config_list = attributes["master_authorized_networks_config"]
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/context/gcp/resources_builders/terraform/container_cluster_builder.py:14]: Shouldn't we use here self._get_known_values?

**Review by Imanuel-Mizrahi — APPROVED** [review]

File: cloudrail/knowledge/context/gcp/resources_builders/terraform/container_cluster_builder.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+from cloudrail.knowledge.context.gcp.resources_builders.terraform.base_gcp_terraform_builder import BaseGcpTerraformBuilder
+
+
+class ContainerClusterBuilder(BaseGcpTerraformBuilder):
+
+    def do_build(self, attributes: dict) -> GcpContainerCluster:
+        name = attributes["name"]
+        location = self._get_known_value(attributes, "location")
+        cluster_ipv4_cidr = self._get_known_value(attributes, "cluster_ipv4_cidr")
+        enable_shielded_nodes = attributes["enable_shielded_nodes"]
```
Comment (Tomerva) [inline cloudrail/knowledge/context/gcp/resources_builders/terraform/container_cluster_builder.py:13]: enable_shielded_nodes is required? if not, use .get

File: tests/knowledge/context/gcp/test_container_cluste.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        self.assertEqual(cluster.location, 'us-central1-a')
+        self.assertTrue(cluster.enable_shielded_nodes)
+        self.assertIsNotNone(cluster.master_authorized_networks_config)
+        self.assertIsNotNone(cluster.master_authorized_networks_config.cidr_blocks)
+        self.assertEqual(cluster.master_authorized_networks_config.cidr_blocks[0].cidr_block, '0.0.0.0/0')
+        self.assertEqual(cluster.master_authorized_networks_config.cidr_blocks[0].display_name, 'office')
+        self.assertIsNotNone(cluster.authenticator_groups_config)
+        self.assertEqual(cluster.authenticator_groups_config.security_group, 'gke-security-groups@indeni.com')
+        if cluster.origin == EntityOrigin.TERRAFORM:
+            self.assertIsNone(cluster.cluster_ipv4_cidr)
```
Comment (Tomerva) [inline tests/knowledge/context/gcp/test_container_cluste.py]: why is it none? there isnt a default value or some way to calculate what the cidr should be?

**Review by Tomerva — CHANGES_REQUESTED** [review]

File: tests/knowledge/context/gcp/test_container_cluste.py (hunk 4 above)
Comment (doronafota) [reply tests/knowledge/context/gcp/test_container_cluste.py]: Screen Shot 2021-11-18 at 14 36 21

i dont think we can calculate what the cidr should be.

**Review by Tomerva — APPROVED** [review]

**Pull request merged**